src/qt/guiutil.cpp

```diff
@@ -857,10 +857,10 @@ void restoreWindowGeometry(const QString& strSetting, const QSize& defaultSize,
 
     parent->resize(size);
     parent->move(pos);
-
-    if ((!pos.x() && !pos.y()) || (QApplication::desktop()->screenNumber(parent) == -1))
-    {
-        QRect screen = QApplication::desktop()->screenGeometry();
+    
+    int screen_number = QApplication::desktop()->screenNumber(parent);
+    QRect screen = QApplication::desktop()->availableGeometry(screen_number);
```

Comment:
What if the result of `availableGeometry()` when `screen_number = 1`?

Reply:
Do you mean -1? If the parameter is -1, it returns the default screen

Reply:
Fixed above comment.